app/src/main/java/uz/direction/noteapp/MainActivity.kt

```diff
@@ -1,11 +1,48 @@
 package uz.direction.noteapp
 
+import android.content.Context
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.Toast
+import uz.direction.noteapp.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
+    companion object{
+        const val MAIN_HEADER_MESSAGE = "header"
+        const val MAIN_TEXT_MESSAGE = "text"
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+
+        val binding = ActivityMainBinding.inflate(layoutInflater)
+        val sharedPref = getSharedPreferences(getString(R.string.shared_preference_key),
+            Context.MODE_PRIVATE)
+
+        setContentView(binding.root)
+
+        binding.headerTitle.setText(sharedPref.getString(getString(R.string.header_key), ""))
+        binding.textValue.setText(sharedPref.getString(getString(R.string.text_key), ""))
+
+        binding.saveButton.setOnClickListener{
+            val intent = Intent(this, ShowNote::class.java)
+            val header = binding.headerTitle.text.toString()
+            val text = binding.textValue.text.toString()
+
+            if ((header == "") || (text == "")){
```

Comment:
in kotlin we have extension functions from box. For example you can replace `header == ""` with `header.isEmpty()`, you can call this function on strings

Reply:
Okay, thanks for explanation